src/cholesky.jl

```diff
@@ -680,10 +680,12 @@ end
 Base.propertynames(F::Cholesky, private::Bool=false) =
     (:U, :L, :UL, (private ? fieldnames(typeof(F)) : ())...)
 
-function Base.:(==)(C1::Cholesky, C2::Cholesky)
+function _equal(C1::Cholesky, C2::Cholesky)
```

Comment:
This would already solve the issue:

```suggestion
function Base.:(==)(C1::C, C2::D) where {C<:Cholesky, D<:Cholesky}
```

and doesn't require that internal function.

Reply:
Ah, indeed, that's fair!